segmentation.py

```diff
@@ -0,0 +1,575 @@
+            self.__prod[m:, m:] += np.outer(Y, Y)
+            self.__weight += 1
+
+    def logconst(self):
+
+        m, n = self.__m, self.__n
+
+        # Note usage of the log-determinant 'trick':
+        #
+        #     log(det(A)) = 2*sum(log(diag(chol(A))))
+        #
+        d = np.diag(linalg.cholesky(self.__prod))
+        w = self.__weight
+
+        # Evaluate the log-normalization constant.
+        # (Equation 8)
+        return special.gammaln(0.5*(w - np.arange(n))).sum() - \
+               n * np.log(d[:m]).sum() - \
+               w * np.log(d[m:] / np.sqrt(w)).sum() - \
+               n * (0.5 * w) * np.log(0.5 * w)
+
+    def param(self):
+        """Return parameters of the posterior distribution."""
+
+        m = self.__m
+        s = linalg.cholesky(self.__prod).transpose()
+        w = self.__weight
+
+        # Compute the parameters of the posterior distribution.
+        return linalg.solve(s[:m, :m], s[:m, m:]), \
+               np.dot(s[:m, :m].transpose(), s[:m, :m]), \
+               np.dot(s[m:, m:].transpose(), s[m:, m:]) / w, \
+               w
+
+    def rand(self):
+
+        m, n = self.__m, self.__n
+
+        s = linalg.cholesky(self.__prod).transpose()
+        w = self.__weight
+
+        # Compute the parameters of the posterior distribution.
+        mu = linalg.solve(s[:m, :m], s[:m, m:])
+        omega = np.dot(s[:m, :m].transpose(), s[:m, :m])
+        sigma = np.dot(s[m:, m:].transpose(), s[m:, m:]) / w
+        eta = w
+
+        # Simulate the marginal Wishart distribution.
+        f = linalg.solve(np.diag(np.sqrt(2.0*random.gamma(
+            (eta - np.arange(n))/2.0))) + np.tril(random.randn(n, n), -1),
+                         np.sqrt(eta)*linalg.cholesky(sigma).transpose())
+        b = np.dot(f.transpose(), f)
+
+        # Simulate the conditional Gauss distribution.
+        a = mu + linalg.solve(linalg.cholesky(omega).transpose(),
+                              np.dot(random.randn(m, n),
+                                     linalg.cholesky(b).transpose()))
+
+        return a, b
+
+
+class Bcdm():
+    """Bayesian change detection model.
+
+    Args:
+        mu (numpy.array): (M x N) location hyper-parameter used when matrix
+                          variate, normal inverse gamma distributions are
+                          initialised during segmentation.
+        omega (numpy.array): (M x M) scale hyper-parameter used when matrix
+                             variate, normal inverse gamma distributions are
+                             initialised during segmentation.
+        sigma (numpy.array): (N x N) dispersion hyper-parameter used when
+                             matrix variate, normal inverse gamma distributions
+                             are initialised during segmentation.
+        eta (float): shape hyper-parameter used when matrix variate, normal
+                     inverse gamma distributions are initialised during
+                     segmentation.
```

Comment:
In the executable code or the documentation? In the executable code we definitely can. 

Not so easily in the documentation. I have just documented each input used to instantiate the object. We could change the input so that it accepts an object which wraps these parameters (like the current MatrixVariateNormalInvGamma object). This would reduce the amount of documentation, in this class, by moving it else where.